### PR TITLE
Extract TXD pool class

### PR DESCRIPTION
**Pull request by TheNormalnij**

### CODE REVIEW

File: Client/game_sa/CTxdPoolSA.cpp [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+    return (*m_ppTxdPoolInterface)->GetObjectIndex(pTxd);
+}
+
+void CTxdPoolSA::RemoveTextureDictonarySlot(std::uint32_t uiTxdId)
+{
+    if (!(*m_ppTxdPoolInterface)->IsContains(uiTxdId))
+        return;
+
+    typedef std::uint32_t(__cdecl * Function_TxdReleaseSlot)(std::uint32_t uiTxdId);
```
Comment (TracerDS) [inline Client/game_sa/CTxdPoolSA.cpp:43]: `using` instead of `typedef`

File: Client/game_sa/CTxdPoolSA.cpp [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+}
+
+bool CTxdPoolSA::IsFreeTextureDictonarySlot(std::uint32_t uiTxdId)
+{
+    return (*m_ppTxdPoolInterface)->IsEmpty(uiTxdId);
+}
+
+std::uint16_t CTxdPoolSA::GetFreeTextureDictonarySlot()
+{
+    return (*m_ppTxdPoolInterface)->GetFreeSlot();
```
Comment (TracerDS) [inline Client/game_sa/CTxdPoolSA.cpp:56]: `const`?

File: Client/mods/deathmatch/logic/CClientModel.cpp [hunk 3]
```diff
@@ -216,7 +216,7 @@ void CClientModel::RestoreDFF(CModelInfo* pModelInfo)
 
 bool CClientModel::AllocateTXD(std::string &strTxdName)
 {
-    uint uiSlotID = g_pGame->GetPools()->AllocateTextureDictonarySlot(m_iModelID - MAX_MODEL_DFF_ID, strTxdName);
+    uint uiSlotID = g_pGame->GetPools()->GetTxdPool().AllocateTextureDictonarySlot(m_iModelID - MAX_MODEL_DFF_ID, strTxdName);
```
Comment (TracerDS) [inline Client/mods/deathmatch/logic/CClientModel.cpp]: `std::uint32_t`

File: Client/mods/deathmatch/logic/CClientModelManager.cpp [hunk 4]
```diff
@@ -73,7 +73,7 @@ int CClientModelManager::GetFirstFreeModelID(void)
 
 int CClientModelManager::GetFreeTxdModelID()
 {
-    ushort usTxdId = g_pGame->GetPools()->GetFreeTextureDictonarySlot();
+    ushort usTxdId = g_pGame->GetPools()->GetTxdPool().GetFreeTextureDictonarySlot();
```
Comment (TracerDS) [inline Client/mods/deathmatch/logic/CClientModelManager.cpp]: `std::uint16_t`

File: Client/sdk/game/CPools.h [hunk 5]
```diff
@@ -107,12 +108,7 @@ class CPools
...
-    virtual uint AllocateTextureDictonarySlot(uint uiSlotID, std::string& strTxdName) = 0;
-    virtual void RemoveTextureDictonarySlot(uint uiTxdID) = 0;
-    virtual bool IsFreeTextureDictonarySlot(uint uiTxdID) = 0;
-
-    virtual ushort GetFreeTextureDictonarySlot() = 0;
-
     virtual CBuildingsPool& GetBuildingsPool() noexcept = 0;
-    virtual CDummyPool&   GetDummyPool() noexcept = 0;
+    virtual CDummyPool&     GetDummyPool() noexcept = 0;
+    virtual CTxdPool&       GetTxdPool() noexcept = 0;
```
Comment (TracerDS) [inline Client/sdk/game/CPools.h:113]: Can you add const reference functions too?

**Review by TracerDS — CHANGES_REQUESTED** [review]

File: Client/game_sa/CTxdPoolSA.cpp (hunk 1 above)
Comment (TheNormalnij) [reply Client/game_sa/CTxdPoolSA.cpp:43]: It's not related to PR.

File: Client/game_sa/CTxdPoolSA.cpp (hunk 2 above)
Comment (TheNormalnij) [reply Client/game_sa/CTxdPoolSA.cpp:56]: It isn't related to PR and GetFreeSlot is not `const`

File: Client/sdk/game/CPools.h (hunk 5 above)
Comment (TheNormalnij) [reply Client/sdk/game/CPools.h:113]: `CTxdPool` has no const methods, const reference will be unused

File: Client/game_sa/CTxdPoolSA.cpp (hunk 1 above)
Comment (TracerDS) [reply Client/game_sa/CTxdPoolSA.cpp:43]: You are adding new functionality wdym

Comment (TheNormalnij) [reply Client/game_sa/CTxdPoolSA.cpp:43]: I don't add a new functionality. This PR only extracts a new class from another class. We don't change any methods here to keep git history clean.

**Comment by Dutchman101** [comment]
This is just a refactor, no changes.

TheNormalnij — Today at 2:59 AM
Only class was extracted, no changes in methods

**Pull request merged**